components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/util/JSONResponseWriter.java

```diff
@@ -115,14 +132,221 @@ private static JsonObject abstractResultToJSONObject(AbstractResult result) thro
+
+                    case EntitlementEndpointConstants.CATEGORY_CODEBASE_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_CODEBASE, getJsonObject(attribute));
+                        break;
+
+                    case EntitlementEndpointConstants.CATEGORY_REQUESTING_MACHINE_URI:
+
+                        jsonResult.add(EntitlementEndpointConstants.CATEGORY_REQUESTING_MACHINE, getJsonObject(attribute));
+                        break;
+                }
+            }
+
+        }
+
         /**
-         * Todo: Category, PolicyIdentifierList
+         * Todo: PolicyIdentifierList
          */
 
         return jsonResult;
     }
 
-    /**
+    /***
+     * Create json object value of an Attribute
+     * @param attribute an element of type Attributes
+     * @return
+     */
+    private static JsonObject getJsonObject(Attributes attribute) {
+        JsonObject jsonObject = new JsonObject();
+        JsonArray jsonArray = new JsonArray();
+        for (Object att : attribute.getAttributes().toArray()) {
+            Attribute atri = (Attribute) att;
+            if (atri.isIncludeInResult()) {
+                JsonObject ele = new JsonObject();
+                if (atri.getId() != null) {
+                    if (xacmlJSONprofileShortFormEnable) {
+                        ele.addProperty(EntitlementEndpointConstants.ATTRIBUTE_ID, uriToShortenForm(atri.getId().toString()));
+                    } else {
+                        ele.addProperty(EntitlementEndpointConstants.ATTRIBUTE_ID, atri.getId().toString());
+                    }
+                }
+                if (atri.getValues() != null) {
+                    for (AttributeValue val : atri.getValues()) {
+                        if (((StringAttribute) val).getValue() != null) {
+                            ele.addProperty(EntitlementEndpointConstants.ATTRIBUTE_VALUE, ((StringAttribute) val).getValue());
+                        }
+                    }
+                }
+                ele.addProperty(EntitlementEndpointConstants.ATTRIBUTE_INCLUDE_IN_RESULT, String.valueOf(atri.isIncludeInResult()));
+                if (atri.getType() != null) {
+                    if (xacmlJSONprofileShortFormEnable) {
+                        ele.addProperty(EntitlementEndpointConstants.ATTRIBUTE_DATA_TYPE, uriToShortenForm(atri.getType().toString()));
+                    } else {
+                        ele.addProperty(EntitlementEndpointConstants.ATTRIBUTE_DATA_TYPE, atri.getType().toString());
+                    }
+                }
+                jsonArray.add(ele);
+            }
+        }
+        jsonObject.add(EntitlementEndpointConstants.ATTRIBUTE, jsonArray);
+        return jsonObject;
+    }
+
+    /***
+     *  This converts the XACML xml data type to simple shorten data type format
+     * @param uriName XACML xml data type uri
+     * @return
+     */
+    private static String uriToShortenForm(String uriName) {
```

Comment:
Need formatting in doc comments and add a newline after the method signature, please.